mypy/applytype.py

```diff
@@ -38,8 +41,11 @@ def apply_generic_arguments(callable: CallableType, types: List[Type],
                     types[i] = value
                     break
             else:
-                msg.incompatible_typevar_value(callable, i + 1, type, context)
-
+                constraints = get_incompatible_arg_constraints(msg, callable.arg_types, type, i + 1)
```

Comment:
Could you please also fix the lint issue -- this line is too long (more than 99 allowed chars).